src/deadline/client/cli/_groups/asset_group.py

```diff
@@ -155,3 +260,146 @@ def asset_download(**args):
     Downloads input manifest of previously submitted job.
     """
     click.echo("download complete")
+
+
+def read_local_manifest(manifest: str) -> BaseAssetManifest:
+    """
+    Read manifets specified by filepath to manifest folder, returns BaseAssetManifest Object
+    """
+    for filename in os.listdir(manifest):
+        if filename.endswith("_input"):
+            # decode text file string into BaseAssetManifest
+            filepath = os.path.join(manifest, filename)
+            with open(filepath, "r") as input_file:
+                manifest_data_str = input_file.read()
+                asset_manifest = decode_manifest(manifest_data_str)
+
+                return asset_manifest
+
+    raise NonValidInputError(
+        f"Specified manifests directory {manifest} does not contain manifest file. "
+    )
+
+
+def clear_S3_mapping(manifest: str):
+    """
+    Clears manifest_s3_mapping file contents if it previously exists.
+    """
+    for filename in os.listdir(manifest):
+        if filename.endswith("manifest_s3_mapping"):
+            # if S3 mapping already exists, clear contents
+            filepath = os.path.join(manifest, filename)
+            with open(filepath, "w") as _:
+                pass
+    return
+
+
+def get_directory_changes():
+    """
+    TODO: Gets the file paths in specified directory if the contents of file have changed compared to a specified snapshot of the respective directory.
+    """
+
+    # root_path = asset_root_manifest.root_path
+
+    # input_paths = []
+    # for root, dirs, files in os.walk(root_path):
+    #     if os.path.samefile(root, manifest):
+    #         dirs[:] = []
+    #         continue
+    #     for filename in files:
+    #         file_path = os.path.join(root, filename)
+    #         input_paths.append(file_path)
+
+    # return input_paths
+
+
+def get_manifest_changes(
+    asset_manager: S3AssetManager,
+    asset_root_manifest: AssetRootManifest,
+    manifest: str,
+    update: bool,
+) -> List[(tuple)]:
+    """
+    Gets the file paths in specified manifest if the contents of file have changed since its last snapshot.
+    """
+    manifest_dir_name = os.path.basename(manifest)
+    root_path = asset_root_manifest.root_path
+    input_paths: List[Path] = []
+
+    asset_manifest = asset_root_manifest.asset_manifest
+    if asset_manifest is None:
+        raise NonValidInputError("Manifest object not found, please check input manifest. ")
+
+    for base_manifest_path in asset_manifest.paths:
+        if base_manifest_path.path.startswith(manifest_dir_name):
+            # skip the manifest folder, or else every upload will need an update after a previous change
+            continue
+        input_paths.append(Path(root_path, base_manifest_path.path))
+
+    return get_file_changes(
+        asset_manager=asset_manager, input_paths=input_paths, root_path=root_path, update=update
+    )
+
+
+def get_file_changes(
```

Comment:
The current name kind of looks like it would mean "get the changes in the file contents" at a glance IMO. How about changing this function to be a little more generic like `find_files_with_status` and a take in a param for the statuses to look for (and pass in `NEW` and `MODIFIED`)?